### PR TITLE
framework/task_manager: clean the source codes in task manager

### PR DESCRIPTION
**Pull request by jeongarmy**
- change the prefix of defined command types from TASKMGT to TASKMGR
- add missing logics and delete unnesassary codes

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [40966bb55d2874685683397f82411cd0f0fc8ff4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [40966bb55d2874685683397f82411cd0f0fc8ff4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [447ba9d9f8b5da596290f4fefdca0fc1153000d0] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [447ba9d9f8b5da596290f4fefdca0fc1153000d0] - Code Rule Check (C++) OK.

File: framework/src/task_manager/task_manager_internal.h [hunk 1]
```diff
@@ -23,22 +23,21 @@
...
-#define TASKMGT_UNICAST                 8
-#define TASKMGT_BROADCAST               9
-#define TASKMGT_SET_UNICAST_CB          10
-#define TASKMGT_SCAN_NAME               11
-#define TASKMGT_SCAN_HANDLE             12
-#define TASKMGT_SCAN_GROUP              13
-#define TASKMGT_UNREGISTER_TASK         14
-#define TASKMGT_SET_BROADCAST_CB        15
+#define TASKMGR_REGISTER_TASK           0
+#define TASKMGR_REGISTER_THREAD         1
```
Comment (jeongchanKim) [inline framework/src/task_manager/task_manager_internal.h]: We don't use this define anymore

File: framework/src/task_manager/task_manager_internal.h [hunk 2]
```diff
@@ -23,22 +23,21 @@
...
+#define TASKMGR_START                   2
+#define TASKMGR_TERMINATE               3
+#define TASKMGR_RESTART                 4
+#define TASKMGR_PAUSE                   5
+#define TASKMGR_RESUME                  6
+#define TASKMGR_UNICAST                 8
```
Comment (jeongchanKim) [inline framework/src/task_manager/task_manager_internal.h]: please check the number sequence

File: framework/src/task_manager/task_manager_register.c [hunk 3]
```diff
@@ -51,19 +51,24 @@ int task_manager_register(char *name, int permission, int timeout)
...
+		if (request_msg.q_name == NULL) {
+			TM_FREE(request_msg.data);
+			return TM_OUT_OF_MEMORY;
+		}
 	}
 
 	status = taskmgr_send_request(&request_msg);
 	if (status < 0) {
+		TM_FREE(request_msg.data);
+		if (request_msg.q_name != NULL) {
```
Comment (jeongchanKim) [inline framework/src/task_manager/task_manager_register.c]: in here, request_msg.q_name cannot be null

Comment (jeongarmy) [reply framework/src/task_manager/task_manager_register.c]: When timeout is not TM_NO_RESPONSE and sending request is failed, it can be NULL.

**Comment by seinfra** [comment]
Target : [37dd40eac1592aa5fb4e1369d6eb2ee511434cf3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [37dd40eac1592aa5fb4e1369d6eb2ee511434cf3] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [623153550cb8eae5177e3c8734a63795c0ebee4e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [623153550cb8eae5177e3c8734a63795c0ebee4e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [afafdad93323eccc32e631aed01e6987a6891f68] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [afafdad93323eccc32e631aed01e6987a6891f68] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [46750447a075508687f99a03857aa0b397ad781f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [46750447a075508687f99a03857aa0b397ad781f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d6b0405268b3a5a1a62995ee58f834caa4831f4b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d6b0405268b3a5a1a62995ee58f834caa4831f4b] - Code Rule Check (C++) OK.

**Pull request merged**